public/app/features/templating/template_srv.ts

```diff
@@ -250,9 +250,13 @@ export class TemplateSrv {
   fillVariableValuesForUrl(params, scopedVars) {
     _.each(this.variables, function(variable) {
       if (scopedVars && scopedVars[variable.name] !== void 0) {
-        params['var-' + variable.name] = scopedVars[variable.name].value;
+        if (scopedVars[variable.name].removeFromUrl !== 1) {
```

Comment:
the change to this function looks a bit strange. I think the check for the option can be done once on the variable instance. The scopedVars will not have this removeFromUrl attribute (unless you specify it somewhere during repeat process). 

I would change this function to not use _.each but use a regular for (let variable of this.variables), and check if (variable.skipUrlSync) continue

Reply:
Well there are two cases here (and they were there before my change).
we're chaking both scoped variable case and non-scoped, see:
```
     _.each(this.variables, function(variable) {
       if (scopedVars && scopedVars[variable.name] !== void 0) {
+        if (scopedVars[variable.name].skipUrlSync) {
+          continue;
+        }
         params['var-' + variable.name] = scopedVars[variable.name].value;
       } else {
+        if (variable.skipUrlSync) {
+          continue;
+        }
         params['var-' + variable.name] = variable.getValueForUrl();
       }
     });
```
So I'm only adding a test for `skipUrlSync` there using `return` to skip as suggested.